src/JobConsumer/Program.cs

```diff
@@ -0,0 +1,316 @@
+        private static readonly TimeSpan DriverTimeout = TimeSpan.FromMinutes(10);
+
+        private static string JobsPath { get; set; }
+        private static string RepoPath { get; set; }
+        private static string DriverPath { get; set; }
+        private static string ServerUrl { get; set; }
+        private static string ClientUrl { get; set; }
+
+        private static string ProcessingPath => Path.Combine(JobsPath, "processing");
+        private static string ProcessedPath => Path.Combine(JobsPath, "processed");
+
+        public static int Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+
+            app.HelpOption("-h|--help");
+            var jobsPath = app.Option("-j|--jobs-path <PATH>", "The path where jobs are created", CommandOptionType.SingleValue).IsRequired();
+            var repoPath = app.Option("-r|--repo-path <PATH>", "The path to the repo being benchmarked", CommandOptionType.SingleValue).IsRequired();
+            var driverPath = app.Option("-d|--driver <PATH>", "The BenchmarksDriver assembly file path", CommandOptionType.SingleValue).IsRequired();
+            var serverUrl = app.Option("-s|--server <URL>", "The server url", CommandOptionType.SingleValue).IsRequired();
+            var clientUrl = app.Option("-c|--client <URL>", "The client url", CommandOptionType.SingleValue).IsRequired();
+
+            app.OnExecuteAsync(async cancellationToken =>
+            {
+                JobsPath = jobsPath.Value();
+                RepoPath = repoPath.Value();
+                DriverPath = driverPath.Value();
+                ServerUrl = serverUrl.Value();
+                ClientUrl = clientUrl.Value();
+
+                var directory = new DirectoryInfo(JobsPath);
+
+                if (!directory.Exists)
+                {
+                    Console.WriteLine($"The path doesn't exist: '{directory.FullName}'");
+                    return -1;
+                }
+
+                if (!File.Exists(DriverPath))
+                {
+                    Console.WriteLine($"The driver could not be found at: '{DriverPath}'");
+                    return -1;
+                }
+
+                // Create the target folders if they don't exist
+                Directory.CreateDirectory(ProcessingPath);
+                Directory.CreateDirectory(ProcessedPath);
+
+                Console.WriteLine("Press enter to exit.");
+
+                while (true)
+                {
+                    // Get oldest file
+                    var nextFile = directory
+                        .GetFiles()
+                        .OrderByDescending(f => f.LastWriteTime)
+                        .FirstOrDefault();
+
+
+                    // If no file was found, wait some time
+                    if (nextFile is null)
+                    {
+                        if (Console.KeyAvailable)
+                        {
+                            if (Console.ReadKey().Key == ConsoleKey.Enter)
```

Comment:
Debugging?

Reply:
I'm going to move both console apps to use Generic Host, but for now this was a quick and dirty alternative to a proper ctrl+c handler.